src/set_theory/game/pgame.lean

```diff
@@ -920,16 +927,19 @@ begin
 end
 using_well_founded { dec_tac := pgame_wf_tac }
 
-theorem add_le_add_left {x y z : pgame} (h : y ≤ z) : x + y ≤ x + z :=
-calc x + y ≤ y + x : add_comm_le
-     ... ≤ z + x : add_le_add_right h
-     ... ≤ x + z : add_comm_le
+instance covariant_class_swap_add_le : covariant_class pgame pgame (swap (+)) (≤) :=
+⟨λ x y z, pgame.add_le_add_right⟩
+
+instance covariant_class_add_le : covariant_class pgame pgame (+) (≤) :=
+⟨λ x y z h, calc x + y ≤ y + x : add_comm_le
+                   ... ≤ z + x : add_le_add_right h _
+                   ... ≤ x + z : add_comm_le⟩
 
 theorem add_congr {w x y z : pgame} (h₁ : w ≈ x) (h₂ : y ≈ z) : w + y ≈ x + z :=
-⟨calc w + y ≤ w + z : add_le_add_left h₂.1
-        ... ≤ x + z : add_le_add_right h₁.1,
- calc x + z ≤ x + y : add_le_add_left h₂.2
-        ... ≤ w + y : add_le_add_right h₁.2⟩
+⟨calc w + y ≤ w + z : add_le_add_left h₂.1 _
```

Comment:
This proof could be simplified a lot if there were an `add_le_add` lemma. However, I can't get the standard one to work here, since it requires a preorder (even though a transitive `≤` should be enough). Maybe a refactor is due?

Reply:
Same thought here. I just didn't want to interfere with Damiano's refactor.